### PR TITLE
Limit deposits for testnet challenge participants

### PR DESCRIPTION
**Pull request by antonyshchenko**
Created a separate ticket to take withdrawals into account when distributing points: https://funkybit.atlassian.net/browse/CHAIN-518

### CODE REVIEW

File: web-ui/src/components/Screens/HomeScreen/DepositModal.tsx [hunk 1]
```diff
@@ -162,15 +165,18 @@ export default function DepositModal({
 
     try {
       if (walletBalanceQuery.status == 'success') {
-        const availableAmount = walletBalanceQuery.data
+        const availableAmount =
+          testnetChallengeDepositLimit !== undefined
+            ? minBigInt(testnetChallengeDepositLimit, walletBalanceQuery.data)
```
Comment (antonyshchenko) [inline web-ui/src/components/Screens/HomeScreen/DepositModal.tsx:170]: Will prevent form submission without telling the reason to the user. Should we inform the user about the limit somehow?

Comment (bholzman) [reply web-ui/src/components/Screens/HomeScreen/DepositModal.tsx:170]: yes, we should definitely inform the user
how about this:
 - if their deposit limit is zero, the Deposit button is disabled with a tooltip that says "Deposits are limited during the Testnet Challenge to previously withdrawn assets."
 - and then in the deposit modal, when the deposit limit exists (even if 0) say something like: "During the Testnet Challenge, deposits are limited to previously withdrawn assets. Current deposit limit: 5,000"

**Review by bholzman — APPROVED** [review]
Looks great!

File: web-ui/src/components/Screens/HomeScreen/DepositModal.tsx (hunk 1 above)
Comment (antonyshchenko) [reply web-ui/src/components/Screens/HomeScreen/DepositModal.tsx:174]: Done, here are screenshots:

Screenshot 2024-09-18 at 13 24 11
Screenshot 2024-09-18 at 13 24 19
Screenshot 2024-09-18 at 13 24 48

**Pull request merged**